internal/gophermart/service/service.go

```diff
@@ -0,0 +1,256 @@
+package service
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/go-resty/resty/v2"
+
+	"github.com/vkupriya/go-gophermart/internal/gophermart/helpers"
+	"github.com/vkupriya/go-gophermart/internal/gophermart/models"
+	"github.com/vkupriya/go-gophermart/internal/gophermart/storage"
+)
+
+type Storage interface {
+	UserAdd(c *models.Config, user models.User) error
+	UserGet(c *models.Config, userid string) (models.User, error)
+	OrderAdd(c *models.Config, userid string, oid string) error
+	OrderGet(c *models.Config, oid string) (models.Order, error)
+	OrdersGet(c *models.Config, userid string) (models.Orders, error)
+	GetAllNewOrders(c *models.Config) (models.Orders, error)
+	UpdateOrder(c *models.Config, order *models.Order) error
+	UserAddAccrual(c *models.Config, order *models.Order) error
+	AccrualWithdraw(c *models.Config, w models.Withdrawal) error
+	WithdrawalsGet(c *models.Config, userid string) (models.Withdrawals, error)
+	BalanceGet(c *models.Config, userid string) (models.Balance, error)
+}
+
+type GophermartService struct {
+	store  Storage
+	config *models.Config
+}
+
+func NewGophermartService(store Storage, cfg *models.Config) *GophermartService {
+	return &GophermartService{
+		store:  store,
+		config: cfg}
+}
+
+func NewStore(c *models.Config) (Storage, error) {
+	ms, err := storage.NewPostgresDB(c)
+	if err != nil {
+		return ms, fmt.Errorf("failed to initialize MemStorage: %w", err)
+	}
+	return ms, nil
+}
+
+func (g *GophermartService) SvcUserAdd(user models.User) error {
+	logger := g.config.Logger
+	err := g.store.UserAdd(g.config, user)
+	if err != nil {
+		return fmt.Errorf("failed to register user %s: %w", user.UserID, err)
+	}
+	logger.Sugar().Debugf("user %s has been registered", user.UserID)
```

Comment:
Используй лучше структурированное логирование, чем `Debugf`, например:

```suggestion
	logger.Sugar().Debugf(
	"user has been registered",
	"user ID", user.UserID,
)
```

Reply:
поменял, да так лучше получается - выводить KV пары.